src/datumaro/plugins/framework_converter.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import logging as log
+from typing import Callable, Optional
+
+import numpy as np
+
+from datumaro.components.annotation import AnnotationType
+from datumaro.components.dataset import Dataset
+
+TASK_ANN_TYPE = {
+    "classification": AnnotationType.label,
+    "multilabel_classification": AnnotationType.label,
+    "detection": AnnotationType.bbox,
+    "instance_segmentation": AnnotationType.polygon,
+    "semantic_segmentation": AnnotationType.mask,
+}
+
+
+class FrameworkConverterFactory:
+    @staticmethod
+    def create_converter(framework):
+        if framework == "torch":
+            return DmTorchDataset
+        elif framework == "tf":
+            return DmTfDataset
+        else:
+            raise ValueError("Unsupported framework")
+
+
+class FrameworkConverter:
+    def __init__(self, dataset, subset, task):
+        self._dataset = dataset
+        self._subset = subset
+        self._task = task
+
+    def to_framework(self, framework, **kwargs):
+        converter_cls = FrameworkConverterFactory.create_converter(framework)
+        return converter_cls(
+            dataset=self._dataset,
+            subset=self._subset,
+            task=self._task,
+            **kwargs,
+        )
+
+
+class _MultiFrameworkDataset:
+    def __init__(
+        self,
+        dataset: Dataset,
+        subset: str,
+        task: str,
+    ):
+        self.dataset = dataset.get_subset(subset)
+        self.task = task
+        self.subset = subset
+        self._ids = [item.id for item in self.dataset]
+
+    def __len__(self) -> int:
+        return len(self.dataset)
+
+    def _gen_item(self, idx: int):
+        item = self.dataset.get(id=self._ids[idx], subset=self.subset)
+        image = item.media.data
+
+        if self.task == "classification":
+            label = [ann.label for ann in item.annotations if ann.type == TASK_ANN_TYPE[self.task]]
+            if len(label) > 1:
+                log.warning(
+                    "Item %s has multiple labels and we choose the first one by default. "
+                    "Please choose task=multilabel_classification for allowing this",
+                    item.id,
+                )
+            label = label[0]
+        elif self.task == "multilabel_classification":
+            label = [ann.label for ann in item.annotations if ann.type == TASK_ANN_TYPE[self.task]]
+        elif self.task in ["detection", "instance_segmentation"]:
+            label = [
+                ann.as_dict() for ann in item.annotations if ann.type == TASK_ANN_TYPE[self.task]
+            ]
+        elif self.task == "semantic_segmentation":
+            masks = [
+                ann.as_class_mask()
+                for ann in item.annotations
+                if ann.type == TASK_ANN_TYPE[self.task]
+            ]
+            label = np.sum(masks, axis=0, dtype=np.uint8)
```

Comment:
I'm worrying that this summation can make problems when there are some overlapped pixels between binary masks. Would you please consider using our `mask_tools.merge_masks()`?

https://github.com/openvinotoolkit/datumaro/blob/cf244a29199b9132f974e9f2d35bd497c555860c/src/datumaro/util/mask_tools.py#L378-L406

Reply:
fixed at cac9e9055749765f6ef306fed825c60a75fb80a2